src/ride/ride.c

```diff
@@ -4540,7 +4540,7 @@ train_ref vehicle_create_train(int rideIndex, int x, int y, int z, int vehicleIn
 {
 	rct_ride *ride = get_ride(rideIndex);
 
-	uint8 trainLayout[42];
+	uint8 trainLayout[RIDE_MAX_CARS_PER_TRAIN];
```

Comment:
Such check would be useful and I intended to add such one in #3179. The problem is this code is duplicated in many places and such a check could be prone to errors, especially when refactoring it some time later on. This should really be refactored now into a small function that would make sure you can only access valid amount of cars in train.